pkg/volume/csi/csi_plugin.go

```diff
@@ -54,12 +108,226 @@ func ProbeVolumePlugins() []volume.VolumePlugin {
 	return []volume.VolumePlugin{p}
 }
 
+// Clones the given map and returns a new map with the given key and value added.
+// Returns the given map, if annotationKey is empty.
+func cloneAndAddAnnotation(
```

Comment:
Consider moving all of this new code to its own file (csi_kubelet_regis.go maybe) to keep csi_plugin.go clean(er).